frontend/src/app/pages/exercises/exercise/shared/simulation/signaller-modal/signaller-modal-interactions/signaller-modal-interactions.component.ts

```diff
@@ -0,0 +1,314 @@
+import type { OnChanges, OnDestroy, OnInit } from '@angular/core';
+import { Component, Input, ViewChild, ElementRef } from '@angular/core';
+import { Store } from '@ngrx/store';
+import type {
+    ExerciseRadiogram,
+    ExerciseSimulationBehaviorType,
+} from 'digital-fuesim-manv-shared';
+import type { UUID } from 'digital-fuesim-manv-shared';
+import {
+    StrictObject,
+    getInformationRequestKeyDetails,
+    isAccepted,
+    isInterfaceSignallerKeyForClient,
+    isUnread,
+} from 'digital-fuesim-manv-shared';
+import { groupBy } from 'lodash-es';
+import type { BehaviorSubject, Observable } from 'rxjs';
+import { Subject, map, takeUntil } from 'rxjs';
+import { ExerciseService } from 'src/app/core/exercise.service';
+import type { SearchableDropdownOption } from 'src/app/shared/components/searchable-dropdown/searchable-dropdown.component';
+import type { HotkeyLayer } from 'src/app/shared/services/hotkeys.service';
+import {
+    Hotkey,
+    HotkeysService,
+} from 'src/app/shared/services/hotkeys.service';
+import type { AppState } from 'src/app/state/app.state';
+import { selectOwnClientId } from 'src/app/state/application/selectors/application.selectors';
+import {
+    createSelectBehaviorStates,
+    selectRadiograms,
+} from 'src/app/state/application/selectors/exercise.selectors';
+import { selectStateSnapshot } from 'src/app/state/get-state-snapshot';
+
+export type InterfaceSignallerInteraction = Omit<
+    SearchableDropdownOption,
+    'backgroundColor' | 'color'
+> & {
+    details?: string;
+    keywords?: string[];
+    hotkey: Hotkey;
+    secondaryHotkey?: Hotkey;
+    requiredBehaviors: ExerciseSimulationBehaviorType[];
+    errorMessage?: string;
+    loading$?: BehaviorSubject<boolean>;
+};
+
+export function setLoadingState(
+    interactions: InterfaceSignallerInteraction[],
+    key: string,
+    loadingState: boolean
+) {
+    interactions
+        .find((information) => information.key === key)
+        ?.loading$?.next(loadingState);
+}
+
+@Component({
+    selector: 'app-signaller-modal-interactions',
+    templateUrl: './signaller-modal-interactions.component.html',
+    styleUrls: ['./signaller-modal-interactions.component.scss'],
+    standalone: false,
+})
+export class SignallerModalInteractionsComponent
+    implements OnInit, OnChanges, OnDestroy
+{
+    @Input()
+    simulatedRegionId?: UUID;
+    @Input()
+    interactions: InterfaceSignallerInteraction[] = [];
+    @Input()
+    primaryActionLabel = '';
+    @Input()
+    showSecondaryButton = true;
+    @Input()
+    filterHotkeyKeys!: string;
+
+    @ViewChild('filterInput')
+    filterInput!: ElementRef;
+
+    private hotkeyLayer: HotkeyLayer | null = null;
+    private filterLayer: HotkeyLayer | null = null;
+    private clientId!: UUID;
+
+    filter = '';
+    filterActive = false;
+    selectedIndex = -1;
+
+    get filteredInteractions() {
+        const lowerFilter = this.filter.toLowerCase();
+
+        return this.interactions.filter(
```

Comment:
This is a little weird in the sense that you cannot filter by multiple phrases by entering multiple words. Maybe you could split the `lowerFilter` in whitespace and then require that all parts match, each using the existing filter logic.

Reply:
Nice idea, added in 9b6fa4db7e2a1e37e3edcc507c87c55ba5cf7607